service.go

```diff
@@ -48,10 +52,16 @@ func NewService(config ServiceConfig) (*Service, error) {
 		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
 	}
 
+	if config.WatchTimeOut.Seconds() == 0 {
+		config.WatchTimeOut = WatchTimeOut
+	}
```

Comment:
We already defined a default via `DegaultConfig`. At this point an error should be returned because the default was overwritten with an invalid value.